src/main/java/com/spleefleague/commands/commands/hat.java

```diff
@@ -0,0 +1,47 @@
+import com.spleefleague.annotations.Endpoint;
+import com.spleefleague.annotations.LiteralArg;
+import com.spleefleague.core.plugin.CorePlugin;
+import com.spleefleague.commands.command.BasicCommand;
+import com.spleefleague.core.player.Rank;
+import org.bukkit.Material;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.PlayerInventory;
+
+public class hat extends BasicCommand {
+
+  public hat(CorePlugin plugin, String name, String usage) {
+    super(plugin, new hatDispatcher(), name, usage, Rank.MODERATOR);
+  }
+
+  @Endpoint
```

Comment:
@Endpoint should specify, that this command is only valid for players (command blocks can't give themselves a hat). See other commands like /tp for example usage.

Reply:
Maybe also add an option to set the hat of other players